extensions/podman/packages/extension/scripts/podman-download.ts

```diff
@@ -176,182 +152,25 @@ export class ShaCheck {
-      console.log(`📔 Downloaded to ${destFile}`);
-    } else {
-      console.log(`⏭️  Skipping ${artifactName} (already downloaded)`);
-    }
-
-    if (!(await this.#shaCheck.checkFile(destFile, msiSha))) {
-      console.warn(`❌ Invalid checksum for ${fileName} downloading again...`);
-      fs.rmSync(destFile);
-      this.#downloadAttempt++;
-      await this.downloadAndCheckSha(tagVersion, fileName, artifactName);
-    } else {
-      console.log(`✅ Valid checksum for ${fileName}`);
-    }
-
-    this.#downloadAttempt = 0;
-  }
-}
-
-export class Podman5DownloadFedoraImage {
-  readonly MAX_DOWNLOAD_ATTEMPT = 3;
-  #downloadAttempt = 0;
-  #octokit: Octokit;
-  #version: string;
-
-  #downloadAndCheck: DownloadAndCheck;
-
-  constructor(
-    readonly version: string,
-    readonly octokit: Octokit,
-    readonly downloadAndCheck: DownloadAndCheck,
-  ) {
-    this.#version = version;
-    this.#octokit = octokit;
-    this.#downloadAndCheck = downloadAndCheck;
-  }
-
-  // For Windows binaries, grab the latest release from GitHub repository
-  async download(arch: string): Promise<void> {
-    if (this.#downloadAttempt >= this.MAX_DOWNLOAD_ATTEMPT) {
-      console.error('Max download attempt reached, exiting...');
-      process.exit(1);
-    }
-
-    const owner = 'containers';
-    const repo = 'podman-machine-wsl-os';
-
-    // now, grab the files
-    const release = await this.#octokit.request('GET /repos/{owner}/{repo}/releases/latest', {
-      owner,
-      repo,
-    });
-
-    let artifactArch;
-    if (arch === 'x64') {
-      artifactArch = 'amd64';
-    } else if (arch === 'arm64') {
-      artifactArch = 'arm64';
-    }
-
-    const artifactName = `${this.#version}-rootfs-${artifactArch}.tar.zst`;
-    const filename = `podman-image-${arch}.tar.zst`;
-    const artifactRelease = release.data.assets.find(asset => asset.name === artifactName);
-
-    if (!artifactRelease) {
-      throw new Error(
-        `Can't find asset with name ${artifactName} to download and verify for podman image from repository ${repo}`,
-      );
-    }
-
-    // tag version
-    const tagVersion = release.data.tag_name;
-    await this.#downloadAndCheck.downloadAndCheckSha(tagVersion, filename, artifactName, owner, repo);
-  }
-}
-
 export class Podman5DownloadMachineOS {
   #version: string;
   #shaCheck: ShaCheck;
   #assetsFolder: string;
+  #diskType: DiskType;
+  #ociRegistryProjectLink: string;
 
   constructor(
     readonly version: string,
     readonly shaCheck: ShaCheck,
     readonly assetsFolder: string,
+    readonly diskType: DiskType,
   ) {
+    this.#diskType = diskType;
     this.#version = version;
     this.#shaCheck = shaCheck;
     this.#assetsFolder = assetsFolder;
+    // Windows uses WSL => machine-os-wsl ; MacOS uses Applehv => machine-os
+    this.#ociRegistryProjectLink = `https://quay.io/v2/podman/${diskType === DiskType.WSL ? 'machine-os-wsl' : 'machine-os'}`;
```

Comment:
IMHO, throw an error in case of unsupported diskType (more robust)
if we add Hyper-V or another diskType it'll default to machine-os repository so I'm not sure it would be expected